WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/WooShipping Create Shipping Labels/WooShipping Package and Rate Selection/WooShippingAddPackageView.swift

```diff
@@ -0,0 +1,254 @@
+        }
+    }
+
+    let dimensionType: DimensionType
+    @Binding var fieldValue: String
+    @FocusState var fieldFocused: Bool
+
+    var body: some View {
+        VStack {
+            HStack {
+                Text(dimensionType.name)
+                    .font(.subheadline)
+                Spacer()
+            }
+            HStack {
+                TextField("", text: $fieldValue)
+                    .keyboardType(.decimalPad)
+                    .font(.body)
+                    .focused($fieldFocused)
+                Text("cm")
+                    .font(.subheadline)
+                    .foregroundStyle(.secondary)
+            }
+            .padding()
+            .roundedBorder(cornerRadius: 8,
+                           lineColor: fieldFocused ? Color(UIColor.wooCommercePurple(.shade60)) : Color(.separator),
+                           lineWidth: fieldFocused ? 2 : 1)
+        }
+        .frame(minHeight: 48)
+    }
+}
+
+#Preview {
+    WooShippingAddPackageView()
+}
+
+extension WooShippingAddPackageView {
+    enum Localization {
+        static let addPackage = NSLocalizedString("wooShipping.createLabel.addPackage.title",
+                                                  value: "Add Package",
+                                                  comment: "Title for the Add Package screen")
+        static let packageType = NSLocalizedString("wooShipping.createLabel.addPackage.packageType",
+                                                   value: "Package type",
+                                                   comment: "Info label for selecting package type")
+        static let cancel = NSLocalizedString("wooShipping.createLabel.addPackage.cancel",
+                                              value: "Cancel",
+                                              comment: "Cancel button in navigation bar to dismiss the screen")
+        static let custom = NSLocalizedString("wooShipping.createLabel.addPackage.custom",
+                                              value: "Custom",
+                                              comment: "Info label for custom package option")
+        static let carrier = NSLocalizedString("wooShipping.createLabel.addPackage.carrier",
+                                               value: "Carrier",
+                                               comment: "Info label for carrier package option")
+        static let saved = NSLocalizedString("wooShipping.createLabel.addPackage.saved",
+                                             value: "Saved",
+                                             comment: "Info label for saved package option")
+        static let box = NSLocalizedString("wooShipping.createLabel.addPackage.box",
+                                           value: "Box",
+                                           comment: "Info label for selected box as a package type")
+        static let envelope = NSLocalizedString("wooShipping.createLabel.addPackage.envelope",
```

Comment:
I also see that in Localizable.strings we have strings like:
`"Envelope" = "Envelope";`
should I just use that or create this new key?
@rachelmcr

Reply:
No, our current approach is the [reverse-DNS style keys](https://github.com/woocommerce/woocommerce-ios/blob/trunk/docs/localization.md#use-unique-reverse-dns-naming-style-keys) so let's stick with that here.